### PR TITLE
document configmap

### PR DESCRIPTION
**Pull request by jcantrill**
This PR documents how to extract files into a configmap

### CODE REVIEW

File: docs/extract_configmap.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+- mountPath: /usr/share/elasticsearch/index_templates
+  name: index-templates
+  ...
+volumes:
+- configMap:
+    defaultMode: 420
+    name: index-templates
+  ...      
+```
+5. Rollout new deployments for each deploymentconfig
```
[the remark this answers is not in the document]
Comment (ewolinetz) [reply docs/extract_configmap.md]: ref: https://docs.openshift.com/container-platform/3.9/install_config/aggregate_logging.html#elasticsearch-rolling-restart

File: docs/extract_configmap.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        "version": "2016.10.12.0"
+      },
+      "date_detection": false,
+      "dynamic_templates": [
+        {
+    ...
+  ```
+  3. Create the configmap
+```
+$ oc create -n $LOGGING_NAMESPACE configmap index-templates --from-file=$DIR
```
Comment (richm) [inline docs/extract_configmap.md]: where `$LOGGING_NAMESPACE` is typically `openshift-logging`, or `logging` for older releases (use `oc get projects` to see which one you have - if you have both, use `openshift-logging`)

EXPERTS: you can also use this method to add your own Elasticsearch index templates - add your own index template .json files to `$DIR`.

File: docs/extract_configmap.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+```
+4. Mount the configmap into each deploymentconfig
+```
+$ oc edit -n $LOGGING_NAMESPACE $ELASTICSEARCH_DC
```
Comment (richm) [inline docs/extract_configmap.md]: `$ELASTICSEARCH_DC` is in the form `dc/logging-es-xxxxx` - Use `oc get -n $LOGGING_NAMESPACE dc` to find the name of your Elasticsearch dc

File: docs/extract_configmap.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+- mountPath: /usr/share/elasticsearch/index_templates
+  name: index-templates
+  ...
+volumes:
+- configMap:
+    defaultMode: 420
+    name: index-templates
+  ...
+```
+5. Follow the formal [procedure for rolloing out new deployments](https://docs.openshift.com/container-platform/3.9/install_config/aggregate_logging.html#elasticsearch-rolling-restart)
```
Comment (richm) [inline docs/extract_configmap.md]: rolling

**Comment by jcantrill** [comment]
@richm updated.

**Pull request merged**